third_party/coro/coro.c

```diff
@@ -103,13 +105,29 @@ coro_init (void)
 
   coro_transfer (new_coro, create_coro);
 
-#if __GCC_HAVE_DWARF2_CFI_ASM && __amd64
-  asm (".cfi_undefined rip");
-#endif
-
-  func ((void *)arg);
-
-  /* the new coro returned. bad. just abort() for now */
+  __asm__(
+	  /*
+	   * Call-chain ends here: we need to invalidate this frame's return
+	   * address to make unwinding stop here.
+	   * Some nuances on x86_64, see:
+	   * https://github.com/libunwind/libunwind/blob/ec171c9ba7ea3abb2a1383cee2988a7abd483a1f/src/x86_64/Gstep.c#L91-L92
+	   * https://github.com/libunwind/libunwind/blob/ec171c9ba7ea3abb2a1383cee2988a7abd483a1f/src/dwarf/Gparser.c#L877
+	   */
+#ifdef ENABLE_BACKTRACE
+	  ".cfi_undefined rip\n"
+	  ".cfi_undefined rbp\n"
+	  ".cfi_return_column rbp\n"
+  #endif /* ENABLE_BACKTRACE */
+	  "\tmovq %0, %%rdi\n"
+	  "\txorq %%rbp, %%rbp\n"
```

Comment:
Nit: It's looks like a little bit messy here, I suggest to move clearing rbp upper, also it will be nice to add comments for each assembly line, like it was done in glibc you referenced.

Reply:
Fixed: move rbp clearing right after CFI assembly and annotated each line with a comment.